### ISSUE TITLE
test failures

### ISSUE THREAD

**Issue by markstos** [disjunction/node-request-plus#3]
Several tests fail `make test`:

```
1) multiple wraps properly processes ENOTFOUND
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      at ontimeout (timers.js:482:11)
      at tryOnTimeout (timers.js:317:5)

2) multiple wraps can be forked to create INDEPENDENT requesters
  - Failed: unexpected success
      at a.then (./test/multipleWrapsSpec.js:140:24)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)
  - Failed: unexpected success
      at rp.then (./test/multipleWrapsSpec.js:102:24)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)

3) promDecorator status_code is set for ENOTFOUND
  - Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
      at ontimeout (timers.js:482:11)
      at tryOnTimeout (timers.js:317:5)

4) promDecorator status_code is set for ETIMEDOUT
  - Expected 'ESOCKETTIMEDOUT' to be 'ETIMEDOUT'.
      at rp.then.catch (./test/promDecoratorSpec.js:100:59)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)

Executed 49 of 49 specs (4 FAILED) in 12 secs.
Makefile:6: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by markstos** [comment]
There are also deprecation warnings:

```
  retryDecorator
    ✓ retries on 503 error
DEPRECATION: done callback received an Error object. Jasmine 3.0 will treat this as a failure
    ✓ fails if max retry reached
DEPRECATION: done callback received an Error object. Jasmine 3.0 will treat this as a failure
    ✓ fails for statusCode=404 without retry
    ✓ uses delay and errorFilter closuers
    ✓ retries timeout
```

**Comment by disjunction** [comment]
Hi @markstos 
thanks for your vivid interest in this library of mine :)

Regarding this and other issues - it feels like i should update the dependencies and in general do some audit. As you may have noticed it hasn't been updated since more than a year and half.

I will try to find the time for it this week, worst case on the weekend.

Cheers

**Comment by disjunction** [comment]
sorry it took me forever :( anyway it's now fixed in 2.0.0, published on npm
